openhands/server/session/agent_session.py

```diff
@@ -134,8 +140,12 @@ async def start(
                     ChangeAgentStateAction(AgentState.AWAITING_USER_INPUT),
                     EventSource.ENVIRONMENT,
                 )
+            success = True
         finally:
             self._starting = False
+            self._monitoring_listener.on_agent_session_start(
+                success, (time.time() - started_at)
+            )
```

Comment:
I think there's a `_status_callback` in here that we should monitor as well

Reply:
Thanks, OK I think I've got that case accounted for now by giving `_create_runtime` a return value. It's strange that `AgentSession.start` will try and continue even if `_create_runtime` failed, but leaving behavior as-is for now, just monitoring.

Reply:
Sorry not exactly what I was thinking, but we can follow up separately